ecgtools/parsers/cesm.py

```diff
@@ -119,6 +135,60 @@ def parse_cesm_history(file):
         return {INVALID_ASSET: file, TRACEBACK: traceback.format_exc()}
 
 
+def parse_cesm_timeseries(file, frequency_dict=frequency_dict):
+    """Parser for CESM Timeseries files"""
+    file = pathlib.Path(file)
+    info = {}
+    try:
+        for stream in CESM_STREAMS:
+            extracted_stream = extract_attr_with_regex(file.stem.lower(), stream.name.lower())
+            if extracted_stream:
+                info['component'] = stream.component
+                info['stream'] = stream.name
+                z = file.stem.split(extracted_stream)
+                info['case'] = z[0].strip('.')
+                info['member_id'] = int(info['case'].split('.')[-1])
+
+                # Use the last part to get variable and time info
+                date_and_variable = z[-1].split('.')
+
+                info['variable'] = date_and_variable[-2]
+                date_range = date_and_variable[-1]
+                start_time, end_time = date_range.split('-')
+                info['start_time'] = parse_date(start_time)
+                info['end_time'] = parse_date(end_time)
+                info['time_range'] = date_range
+                break
+        with xr.open_dataset(file, chunks={}, decode_times=False) as ds:
+
+            # Get the long name from dataset
+            info['long_name'] = ds[info['variable']].attrs.get('long_name')
+
+            # Grab the units of the variable
+            info['units'] = ds[info['variable']].attrs.get('units')
+
+            # Set the default of # of vertical levels to 1
+            info['vertical_levels'] = 1
+
+            try:
+                info['vertical_levels'] = ds[ds.cf['vertical'].name].size
+            except (KeyError, AttributeError, ValueError):
+                pass
+
+            try:
+                info['frequency'] = ds.attrs['time_period_freq']
+
+            except AttributeError:
+                warnings.warn('Using the default frequency definitions')
+                info['frequency'] = frequency_dict[info['stream']]
```

Comment:
This will fail when `info['stream']` doesn't exist in `frequency_dict`. What should we default to when `info['stream']` doesn't exist in `frequency_dict`?

Reply:
This might be good to tie in together then... and just use the defaults (with a given warning for the frequency). Then, if the user wants to include something beyond the defaults, they setup their dictionary in a similar fashion. This would be good to include for the history files one too, since it would be nice to use CESM1 too...

Reply:
Alrighty - I went ahead an merged the dictionaries together/tested on the pinatubo timeseries files...